java-checks/src/test/files/checks/AssertionsInTestsCheck/Junit4.java

```diff
@@ -144,4 +148,18 @@ public void test_fail() {
 
   }
 
+  interface I {
+    default String returnFoo() {
+      return "Foo";
+    }
+  }
+
+  public class ITest implements I {
+    @Test
+    public void testSuper() {
+      assertThat(I.super.returnFoo(), equalTo("Foo"));
```

Comment:
Please use fully qualified names of these static methods if you want to keep this new case. 

However, I'm not sure this is wanted. You have a dedicated unit test related to the fixed semantic. So this is just an extra case which have no reasons NOT to be compliant. It was non-compliant because of a bug in the semantic... which is somehow not directly related to this rule. Some other rule could have been impacted, and we don't want to add test cases for all of them.